src/main/scala/introcourse/level07/ServiceExercises.scala

```diff
@@ -0,0 +1,87 @@
+package introcourse.level07
+
+import cats.effect.IO
+import org.http4s._
+import org.http4s.dsl.io._
+
+@SuppressWarnings(Array("org.wartremover.warts.Any"))
+object Service {
+  /* Our service is going to return a list of user names. Let's start by writing
+   * a function that takes in a list of users, and returns the names of all users,
+   * separated by new lines.
+   *
+   * Hints:
+   * - The newline character is represented as "\n"
+   * - You can use foldLeft or mkString to concatenate the list items
+   */
+  def userNames(users: List[User]): String = ???
+
+  /* The service already has ping endpoint.
+   *
+   * You can try it by running the service:
+   * > sbt run
+   *
+   * Try calling the service at http://localhost:8080/ping
+   *
+   * Look at the status parameter of the Response class: `Status.Ok`.
+   * This is an implemented via an ADT. Why?
+   */
+  def ping(): IO[Response[IO]] = IO(Response(Status.Ok).withEntity("pong"))
```

Comment:
Maybe move this up above the `userNames` so that we can use this as a test to make sure they can get the service running and `curl` against it. More as a test setup validation step.

Reply:
Good idea. I'll shuffle things around a tad. I also realized I have a bit more time for the exercise, so maybe I can leave a few more spots open to implement..